library/core/src/main/java/com/google/android/exoplayer2/source/ads/AdsMediaSource.java

```diff
@@ -332,7 +332,7 @@ private void maybeUpdateAdMediaSources() {
   private void maybeUpdateSourceInfo() {
     @Nullable Timeline contentTimeline = this.contentTimeline;
     if (adPlaybackState != null && contentTimeline != null) {
-      if (adPlaybackState.adGroupCount == 0) {
+      if (adPlaybackState.adGroupCount == 0 || contentTimeline.getPeriodCount() > 1) {
```

Comment:
This is a good idea, so we ignore csai ads for all multi-period streaming, whether live or vod.